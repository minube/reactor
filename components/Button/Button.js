import {
  bool, func, node, number, oneOfType, string,
} from 'prop-types';
import React from 'react';
import { View } from 'react-native';

import { LAYOUT, THEME } from '../../common';
import Activity from '../Activity';
import Icon from '../Icon';
import Text from '../Text';
import Touchable from '../Touchable';
import styles, { REGULAR_SIZE } from './Button.style';

const { COLOR: { BASE, TEXT_LIGHTEN, WHITE } } = THEME;

const Button = ({
  activity, children, color, contained, disabled, icon, onPress, outlined,
  responsive, rounded, shadow, small, title,
  isSolid = contained && !outlined, // eslint-disable-line
  rippleColor = isSolid ? undefined : color, // eslint-disable-line
  ...inherit
}) => (
  <View style={inherit.style}>
    <Touchable
      containerBorderRadius={rounded ? REGULAR_SIZE / 2 : undefined}
      onPress={disabled ? undefined : onPress}
      rippleColor={isSolid && color === WHITE ? BASE : rippleColor}
      style={[styles.touchable, rounded && styles.rounded]}
    >
      <View
        style={[
          styles.container,
          styles.regular,
          styles.row,
          // -- Layout
          small && styles.small,
          responsive && !small && !LAYOUT.VIEWPORT.REGULAR && !LAYOUT.VIEWPORT.LARGE && styles.small,
          rounded && styles.rounded,
          ((!title && !children) || (!contained && !outlined)) && styles.squared,

          // -- Color
          isSolid && { backgroundColor: color || TEXT_LIGHTEN },
          isSolid && shadow && !disabled && styles.shadow,
          isSolid && disabled && styles.disabled,
          outlined && styles.outlined,
          outlined && { borderColor: color || TEXT_LIGHTEN },
          !isSolid && disabled && styles.disabledOpacity,
        ]}
      >
        { activity && <Activity color={isSolid ? WHITE : color || TEXT_LIGHTEN} type="small" /> }
        { icon && !activity && <Icon value={icon} size={inherit.iconSize} /> }
        <View style={[styles.row, (activity || icon) && (title || children) && styles.textMarginLeft]}>
          { title
            && (
            <Text
              color={isSolid && color === WHITE ? TEXT_LIGHTEN : (isSolid ? WHITE : color)} // eslint-disable-line
              style={[
                styles.text,
                small && styles.textSmall,
                responsive && !small && styles.textSmall,
                (activity || icon) && styles.textMarginLeft,
              ]}
            >
              {title}
            </Text>
            ) }
          { children }
        </View>
      </View>
    </Touchable>
  </View>
);

Button.propTypes = {
  activity: bool,
  children: node,
  contained: bool,
  color: string,
  disabled: bool,
  icon: oneOfType([number, string]),
  onPress: func,
  outlined: bool,
  responsive: bool,
  rounded: bool,
  shadow: bool,
  small: bool,
  title: string,
};

Button.defaultProps = {
  activity: false,
  children: undefined,
  contained: true,
  color: undefined,
  disabled: false,
  icon: undefined,
  onPress: undefined,
  outlined: false,
  responsive: false,
  rounded: false,
  shadow: false,
  small: false,
  title: undefined,
};

export default Button;
