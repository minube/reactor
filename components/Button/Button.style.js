import { Platform, StyleSheet } from 'react-native';

import { STYLE, THEME } from '../../common';

const {
  BORDER_RADIUS, BUTTON, COLOR, FONT, UNIT,
} = THEME;

export default StyleSheet.create({

  touchable: {
    margin: UNIT / 2,
    borderRadius: BORDER_RADIUS,
  },

  container: StyleSheet.flatten([
    STYLE.ROW,
    STYLE.CENTERED,
    {
      backgroundColor: COLOR.TEXT,
      height: BUTTON.HEIGHT,
      paddingHorizontal: UNIT * 3,
      borderRadius: BORDER_RADIUS,
      ...Platform.select({
        web: { width: 'fit-content' },
      }),
    },
  ]),

  accent: {
    backgroundColor: COLOR.ACCENT,
  },

  activity: {
    marginRight: UNIT,
  },

  floating: {
    borderRadius: BUTTON.FLOATING_HEIGHT / 2,
    height: BUTTON.FLOATING_HEIGHT,
    paddingHorizontal: 0,
    width: BUTTON.FLOATING_HEIGHT,
  },

  disabled: {
    backgroundColor: COLOR.BORDER,
    borderWidth: 0,
  },

  flat: {
    backgroundColor: COLOR.WHITE,
    borderWidth: UNIT * 0.1,
    borderColor: COLOR.BORDER,
  },

  primary: {
    backgroundColor: COLOR.PRIMARY,
  },

  rounded: {
    borderRadius: BUTTON.HEIGHT / 2,
  },

  small: {
    height: BUTTON.SMALL_HEIGHT,
    paddingHorizontal: BUTTON.SMALL_HEIGHT / 3,
  },

  text: {
    color: COLOR.WHITE,
    fontSize: FONT.SIZE.SMALL,
  },

  textDisabled: {
    opacity: 0.75,
  },

  textFlat: {
    color: COLOR.TEXT_LIGHTEN,
  },

  textSmall: {
    fontSize: FONT.SIZE.TINY,
  },

  icon: {
    width: FONT.SIZE.REGULAR,
    height: FONT.SIZE.REGULAR,
    marginRight: UNIT,
  },

  iconFloating: {
    width: FONT.SIZE.LARGE,
    height: FONT.SIZE.LARGE,
  },
});
