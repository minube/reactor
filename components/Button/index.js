export { default } from './Button'; // eslint-disable-line
