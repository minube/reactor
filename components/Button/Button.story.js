import React from 'react';
import { Text } from 'react-native';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean, color, object } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import { LayoutView, Price } from '../';
import Button from './Button';

const styles = {
  button: {
    paddingHorizontal: 10,
    backgroundColor: 'green',
  },
}
const _onPress = () => {};

storiesOf('✅ Button', module)
  .add('default', () => (
    <Button />
  ))
  .add('title', () => (
    <Button title="Press me" onPress={_onPress} />
  ))
  .add('icon', () => (
    <Button icon="closeContrast" onPress={_onPress} />
  ))
  .add('icon & title', () => (
    <Button icon="closeContrast" title="Press me" onPress={_onPress} />
  ))
  .add('contained (false)', () => (
    <Button contained={false} title="Press me" onPress={_onPress} />
  ))
  .add('outlined', () => (
    <Button outlined title="Press me" onPress={_onPress} />
  ))
  .add('activity', () => (
    <Button activity title="Press me" onPress={_onPress} />
  ))
  .add('children', () => (
    <Button>
      <Price headline level={6} color="white" value={19.95} symbol="$" onPress={_onPress} />
    </Button>
  ))
  .add('color', () => (
    <Button color="red" title="Press me" onPress={_onPress} />
  ))
  .add('color (COLOR.WHITE)', () => (
    <Button color={THEME.COLOR.WHITE} title="Press me" onPress={_onPress} />
  ))
  .add('disabled', () => (
    <Button disabled title="Press me" />
  ))
  .add('responsive', () => (
    <Button responsive title="Press me" onPress={_onPress} />
  ))
  .add('rounded', () => (
    <Button rounded title="Press me" onPress={_onPress} />
  ))
  .add('small', () => (
    <Button small title="Press me" onPress={_onPress} />
  ))
  .add('shadow', () => (
    <Button shadow title="Press me" onPress={_onPress} />
  ))
  .add('⚡ onPress', () => (
    <Button onPress={action('Button.onPress()')} title="Press me" />
  ))
  .add('style', () => (
    <Button title="Press me" style={styles.button} onPress={_onPress} />
  ))
  .add('🏀 Playground', () => (
    <Button
      activity={boolean('activity', false)}
      color={color('color', undefined)}
      contained={boolean('contained', true)}
      disabled={boolean('disabled', false)}
      icon={text('icon', undefined)}
      onPress={action('Button.onPress()')}
      outlined={boolean('outlined', false)}
      responsive={boolean('responsive', false)}
      rounded={boolean('rounded', false)}
      small={boolean('small', false)}
      shadow={boolean('shadow', false)}
      title={text('title', undefined)}
      style={object('style', styles.button)}
    />
  ));

