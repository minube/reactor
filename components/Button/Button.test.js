import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { Price } from '../';
import Button from './Button';

describe('<Button>', () => {
  it('renders', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {title}', () => {
    const tree = renderer.create(<Button title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {icon}', () => {
    const tree = renderer.create(<Button icon="apps" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {icon} & {title}', () => {
    const tree = renderer.create(<Button icon="apps" title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {flat}', () => {
    const tree = renderer.create(<Button flat title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {accent}', () => {
    const tree = renderer.create(<Button accent title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {activity}', () => {
    const tree = renderer.create(<Button activity title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {children}', () => {
    const tree = renderer.create(<Button><Price value={10} /></Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Button color="red" title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<Button disabled title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {responsive}', () => {
    const tree = renderer.create(<Button responsive title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {rounded}', () => {
    const tree = renderer.create(<Button rounded title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {small}', () => {
    const tree = renderer.create(<Button small title="Press me" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onPress}', () => {
    // @TODO: Test <Button /> {onPress}
  });

  it('when {style}', () => {
    const tree = renderer.create(<Button title="Press me" style={{ backgroundColor: 'orange' }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

