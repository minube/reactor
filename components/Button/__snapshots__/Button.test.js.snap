// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  />
</View>
`;

exports[`<Button> when {accent} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#f69a37",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {activity} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#ffffff"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "marginRight": 10,
        }
      }
      type="small"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {children} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
            },
            false,
            false,
            Object {
              "fontWeight": "bold",
            },
            false,
            Object {
              "fontSize": 16,
              "lineHeight": 20.8,
            },
            false,
            false,
            Object {
              "fontSize": 20,
              "lineHeight": 26,
            },
            false,
            false,
            false,
            false,
            false,
            Object {},
          ]
        }
      >
        10
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontFamily": undefined,
            },
            false,
            false,
            false,
            false,
            Object {
              "fontSize": 16,
              "lineHeight": 20.8,
            },
            false,
            false,
            Object {
              "fontSize": 20,
              "lineHeight": 26,
            },
            false,
            false,
            false,
            false,
            false,
            Object {
              "transform": Array [
                Object {
                  "scale": 0.8,
                },
              ],
            },
          ]
        }
      >
        â‚¬
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Button> when {color} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {disabled} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#dbdbdb",
        "borderRadius": 3,
        "borderWidth": 0,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "opacity": 0.75,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {flat} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#dbdbdb",
        "borderRadius": 3,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#9b9b9b",
            "fontSize": 14,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {icon} & {title} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../components/Icon/assets/apps.png",
        }
      }
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 16,
          "marginRight": 10,
          "width": 16,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {icon} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 30,
      "height": 60,
      "margin": 5,
      "paddingHorizontal": 0,
      "width": 60,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 30,
        "display": "flex",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "center",
        "paddingHorizontal": 0,
        "width": 60,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../components/Icon/assets/apps.png",
        }
      }
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`<Button> when {responsive} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
            "lineHeight": 16.8,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {rounded} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 22.5,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 22.5,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {small} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 12,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {style} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "orange",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<Button> when {title} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 3,
      "margin": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#4a4a4a",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#4a4a4a",
            "fontFamily": undefined,
          },
          false,
          Object {
            "fontWeight": "600",
          },
          false,
          false,
          Object {
            "fontSize": 16,
            "lineHeight": 20.8,
          },
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          Object {
            "color": "#ffffff",
            "fontSize": 14,
          },
        ]
      }
    >
      Press me
    </Text>
  </View>
</View>
`;
